### PR TITLE
Emit reloc patch

### LINKED ISSUE #5986: Adding -emit-relocs for use with Post-Build tools.
Hello! I am currently developing a wrapper of the [PSP Toolchain for Zig](https://github.com/NT-Bourgeois-Iridescence-Technologies/Zig-PSP), in order to be able to develop using Zig on the Sony PlayStation Portable. Currently I aim to use no external dependencies. The last external dependency to resolve is LLD.

In order to satisfy post-build tools for this platform, users need relocation sections in the final built ELF file (in order to reformat into Sony's proprietary loading specifications). I have all of my building through a `build.zig` script and use LLD for the `-emit-relocs` flag, which is the only flag missing from zig's default linking options. Since zig internally uses LLD, I believe it would be logical to nix this requirement by using a linker flag.

`-emit-relocs` is an option which allows the emission of relocation sections for use by post-build tools. This might also be a useful feature for other targets such as ARM Embedded Consoles such as GBA, DS, and 3DS that would likely have similar tools required. 

The goal is to be able to work with the user only having the zig compiler and linker available, and simply just including the toolchain in their project with no externals. Seeing that it's one flag, I would like to propose adding -emit-relocs as a conditional flag, and I'm happy to create a patch like what was done in #3981. 

Let me know your thoughts, and I'd be happy to continue this discussion.

### PR DESCRIPTION
**Pull request by IridescentRose**
This small patch adds relocation support behind a flag (-emit-relocs) and resolves #5986. This is a valuable addition for those using post-build tools.

### CODE REVIEW

File: src/link/Elf.zig [hunk 1]
```diff
@@ -1364,6 +1365,10 @@ fn linkWithLLD(self: *Elf, comp: *Compilation) !void {
     if (self.base.options.eh_frame_hdr) {
         try argv.append("--eh-frame-hdr");
     }
+    
+    if (self.base.options.emit_relocs) {
+        try argv.append("-emit-relocs");
```
Comment (LemonBoy) [inline src/link/Elf.zig:1370]: Double slash here according to the [man page](https://manpages.debian.org/experimental/lld-10/ld.lld-10.1.en.html)

File: lib/std/build.zig [hunk 2]
```diff
@@ -2075,6 +2076,9 @@ pub const LibExeObjStep = struct {
         if (self.link_eh_frame_hdr) {
             try zig_args.append("--eh-frame-hdr");
         }
+        if(self.link_emit_relocs){
+            try zig_args.append("-emit-relocs");
```
Comment (LemonBoy) [inline lib/std/build.zig:2080]: Long options require two `-`

File: lib/std/build.zig [hunk 3]
```diff
@@ -2075,6 +2076,9 @@ pub const LibExeObjStep = struct {
         if (self.link_eh_frame_hdr) {
             try zig_args.append("--eh-frame-hdr");
         }
+        if(self.link_emit_relocs){
```
Comment (LemonBoy) [inline lib/std/build.zig:2079]: Run `zig fmt` on the files you modify

Comment (IridescentRose) [reply lib/std/build.zig]: Apologies - this is my first time contributing

File: src/link/Elf.zig (hunk 1 above)
Comment (IridescentRose) [reply src/link/Elf.zig]: Fixed this and the other issue in the newest commit

File: lib/std/build.zig (hunk 2 above)
Comment (IridescentRose) [reply lib/std/build.zig]: Fixed this and the other issue in the newest commit

File: src/main.zig [hunk 4]
```diff
@@ -838,6 +840,8 @@ fn buildOutputType(
                         function_sections = true;
                     } else if (mem.eql(u8, arg, "--eh-frame-hdr")) {
                         link_eh_frame_hdr = true;
+                    } else if (mem.eql(u8, arg, "-emit-relocs")) {
```
Comment (LemonBoy) [inline src/main.zig:843]: Double `-` here too

File: lib/std/build.zig [hunk 5]
```diff
@@ -2075,6 +2076,9 @@ pub const LibExeObjStep = struct {
         if (self.link_eh_frame_hdr) {
             try zig_args.append("--eh-frame-hdr");
         }
+        if(self.link_emit_relocs) {
```
Comment (LemonBoy) [inline lib/std/build.zig:2079]: Missing space between `if` and the parenthesis

File: src/main.zig (hunk 4 above)
Comment (IridescentRose) [reply src/main.zig:843]: Good catch 👍

File: lib/std/build.zig (hunk 3 above)
Comment (LemonBoy) [reply lib/std/build.zig]: No problem, this is what code reviews are for :)

**Pull request merged**